handlers/aws/handler.py

```diff
@@ -4,36 +4,114 @@
 
 import os
 
-from sqs_trigger import _handle_sqs_event
-from utils import _enrich_event, _get_trigger_type
+import elasticapm  # noqa: F401
+from elasticapm.contrib.serverless.aws import capture_serverless  # noqa: F401
+from sqs_trigger import _handle_sqs_continuation, _handle_sqs_event
+from utils import from_s3_uri_to_bucket_name_and_object_key, get_trigger_type
 
-from shippers import CommonShipper, ShipperFactory
+from share import Config, ElasticSearchOutput, logger, parse_config
+from shippers import CommonShipper, CompositeShipper, ShipperFactory
+from storage import CommonStorage, StorageFactory
 
+_completion_grace_period: int = 60000
 
+
+@capture_serverless()
 def lambda_handler(lambda_event, lambda_context):
-    index: str = os.getenv("ES_INDEX")
-    shipper: CommonShipper = ShipperFactory.create(
-        target="elasticsearch",
-        hosts=os.getenv("ES_HOSTS").split(","),
-        username=os.getenv("ES_USERNAME"),
-        password=os.getenv("ES_PASSWORD"),
-        scheme=os.getenv("ES_SCHEME"),
-        index=index,
-    )
+    try:
+        trigger_type: str = get_trigger_type(lambda_event)
+        logger.info("trigger", extra={"type": trigger_type})
+    except Exception as e:
+        logger.exception("exception raised", exc_info=e)
+        return str(e)
 
     try:
-        trigger_type: str = _get_trigger_type(lambda_event)
+        if trigger_type == "self_sqs":
+            payload = lambda_event["Records"][0]["messageAttributes"]
+            config_yaml = payload["config"]["stringValue"]
+            config: Config = parse_config(config_yaml)
+            lambda_event["Records"][0]["eventSourceARN"] = payload["originalEventSource"]["stringValue"]
+
+        else:
+            config_file: str = os.getenv("S3_CONFIG_FILE")
+            if config_file is None:
+                return "empty S3_CONFIG_FILE env variable"
+
+            bucket_name, object_key = from_s3_uri_to_bucket_name_and_object_key(config_file)
+            logger.info("config file", extra={"bucket_name": bucket_name, "object_key": object_key})
+
+            config_storage: CommonStorage = StorageFactory.create(
+                storage_type="s3", bucket_name=bucket_name, object_key=object_key
+            )
+
+            config_yaml = config_storage.get_as_string()
+            logger.debug("config", extra={"yaml": config_yaml})
+
+            config: Config = parse_config(config_yaml)
```

Comment:
Nit. For readability move the handling of each trigger type to their own methods. This may also help to avoid big try/except blocks.